[tingxuanp] iP

tingxuanp opened the pull request:
# Bestie
## The ultimate companion to ensure you never miss another deadline

> For every minute spent in organising, an hour is earned.  - Benjamin Franklin 

Bestie tracks your upcoming todos, deadlines and events. It's 
- Text-based
- Easy to learn
- fast **_SUPER FAST_** to use

Installing and starting Bestie is simple!

All you need to do is:
1. Download the [ip.jar file](https://github.com/tingxuanp/ip/releases/tag/A-Jar).
2. Open a command window in the JAR folder. 
3. Run the command `java -jar "Bestie.jar"` run the command in the same folder as the jar file.
4. Start keying in your tasks and let Bestie manage them for you! 😆 

Still not convinced?
Here are some of Bestie's features:

- [ ] Managing todos 
- [ ] Managing deadlines
- [ ] Tracking events 

If you are a Java programmer, you can use it to practise Java too. Here is the `main` method:
``` Java
public static void main(String[] args) {
        new Bestie("bestie.txt").run();
}
```

Download Bestie today!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

tingxuanp committed:
Rename chatbot and main file

tingxuanp committed:
Implemented skeletal version of chatbot

tingxuanp committed:
Implemented chatbot echo

tingxuanp committed:
Implemented ability to store and display text entered by user

tingxuanp committed:
Created Task class and added task formatting

tingxuanp committed:
Implemented ability to mark tasks as done and undone

tingxuanp committed:
Added Todo, Deadline and Event classes, added support for tracking 3 task types

tingxuanp committed:
Added some simple automated Text UI tests

tingxuanp committed:
Corrected expected output

tingxuanp committed:
Added more automated Text UI tests

tingxuanp committed:
Added some error handling

tingxuanp committed:
Add support for deleting tasks from the list

tingxuanp committed:
Remove unnecessary import file

tingxuanp committed:
Add ability to save tasks in hard drive

tingxuanp committed:
Improve chatbot response

tingxuanp committed:
Fix incorrect formatting

tingxuanp committed:
Implement error handling for mark, unmark, delete commands

tingxuanp committed:
Merge branch 'branch-Level-7'

tingxuanp committed:
Implement ability to read dates

tingxuanp committed:
Add ability to understand time

tingxuanp committed:
Deadline.java and Event.java: Add error handling

tingxuanp committed:
Improve format

tingxuanp committed:
Merge branch 'branch-Level-8'

tingxuanp committed:
Make code more OOP

tingxuanp committed:
Organise class into packages and fix formatting issues

tingxuanp committed:
Merge branch 'add-gradle-support'

tingxuanp committed:
Package App as a JAR file

tingxuanp committed:
Add Javadoc comments

tingxuanp committed:
Follow Java coding standard

tingxuanp committed:
Implement ability to find tasks

tingxuanp committed:
Merge branch 'branch-A-JavaDoc'

tingxuanp committed:
Merge branch 'branch-A-CodingStandard'

tingxuanp committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/bestie/Task.java

tingxuanp committed:
Add missing Javadoc comments

tingxuanp committed:
Add tests for TaskList.java and Deadline.java

tingxuanp committed:
Implement checkstyle and fix style errors

tingxuanp committed:
Add basic GUI to chatbot

tingxuanp committed:
Merge branch 'branch-Level-10'

tingxuanp committed:
Add Javadoc comments and clean up code

WeiHanWong commented on src/main/java/bestie/MainWindow.java:
```diff
@@ -0,0 +1,56 @@
+
+    private Bestie bestie;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/Me.png"));
```
Resource should be final

WeiHanWong commented on src/main/java/bestie/Ui.java:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * Prints goodbye message when user exits the chatbot.
+     */
+    public String byeBye() {
```
methods should be verb

WeiHanWong reviewed (commented):
LGTM. Just some nits to fix.

jewiteo commented on src/main/java/bestie/Ui.java:
```diff
@@ -0,0 +1,149 @@
+     */
+    public String showTaskMarked(Task task) {
+        return ("Awesome work! I've marked this task as done.\n"
+                + "  " + task.toString());
```
The whitespace could be added to after the newline for better clarity

jewiteo commented on src/main/java/bestie/Ui.java:
```diff
@@ -0,0 +1,149 @@
+     */
+    public String showTaskUnmarked(Task task) {
+        return ("Noted! I've marked this task as not done yet:\n"
+                + "  " + task.toString());
```
Same as for showTaskMarked

jewiteo commented on src/main/java/bestie/Storage.java:
```diff
@@ -0,0 +1,118 @@
+        // want to save tasks to bestie.txt file
+        try {
+            File f = new File(this.filePath);
+            FileWriter fw = new FileWriter(f); // f or file path?
```
Unnecessary comment should be removed

jewiteo commented on src/main/java/bestie/task/Event.java:
```diff
@@ -0,0 +1,66 @@
+        this.end = end;
+        DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        this.parsedStart = LocalDateTime.parse(start, inputFormat);
+        this.parsedEnd = LocalDateTime.parse(end, inputFormat);
```
Code could be separated with a blank line such that one portion is for parsing input and the other portion is for formatting the output for better clarity since the code is long

jewiteo reviewed (commented):
The code looks good, with minimal areas to improve on for better clarity

tingxuanp committed:
Update gradle file for new release

tingxuanp committed:
Improve code quality

tingxuanp committed:
Revert "Improve code quality"

This reverts commit f21dcff26d365342ea142549819c07f894476d5c.

tingxuanp committed:
Improve code quality

Some comments are unnecessary and repeat information already obvious
from the code. Some comments also explain the "how" of the code, and
not the "what" and "why".

This reduces the code quality of my code.

Let's improve the naming of variables and methods, avoid
complicated expressions and remove unnecessary comments.

Improving the names of variables and code structure allows my code to
be more self-explanatory thus reducing the need for unnecessary
comments.

tingxuanp committed:
Add assertions to document assumptions in code

tingxuanp committed:
Resolve merge conflicts

tingxuanp committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

tingxuanp committed:
Merge pull request #2 from tingxuanp/branch-A-CodeQuality

Merge Code Quality branch

tingxuanp committed:
Merge branch 'master' of https://github.com/tingxuanp/ip

* 'master' of https://github.com/tingxuanp/ip:
  Resolve merge conflicts
  Add assertions to document assumptions in code
  Improve code quality
  Revert "Improve code quality"
  Improve code quality

tingxuanp committed:
Merge branch 'master' into branch-A-Assertions

tingxuanp committed:
Merge pull request #4 from tingxuanp/branch-A-Assertions

Merge branch-A-assertions into branch 'master'

tingxuanp committed:
Set up CI

tingxuanp committed:
Add a way to attach priority to tasks

Users currently are not able to sort and tag their tasks based on
its priority, which may reduce user experience.

Let's,
* add a feature to allow users to state the priority of their tasks
using the /priority command attached with each task
* add a feature where users can find all lists of a particular priority
using the "priority" command

This allows users to attach a priority to each task and quickly
obtain a list of all tasks of a certain priority.

tingxuanp committed:
Merge branch 'BCD-Extension'

tingxuanp committed:
Fix bug, allow error messages to be displayed

tingxuanp committed:
Add more personality

tingxuanp committed:
Merge branch 'branch-A-Personality'

tingxuanp committed:
Add more error handling for command format errors

tingxuanp committed:
Add more tests and Javadoc comments

tingxuanp committed:
Add more Javadoc comments

tingxuanp committed:
Add Javadoc comments and Ui tests

tingxuanp committed:
Allow resizing and change profile picture to improve GUI

tingxuanp committed:
Fix minor bug and add Ui picture

tingxuanp committed:
Update README.md

tingxuanp committed:
Fix minor mistakes in README.md

tingxuanp committed:
Fix minor style improvements

tingxuanp committed:
Add Parser classes to improve use of SLAP

tingxuanp committed:
Add more error handling, incorporate SLAP in parser, add more tests

tingxuanp committed:
Fix checkstyle errors

tingxuanp committed:
Fix CI Date Parser compatibility bug

tingxuanp committed:
Fix runtest shell script bug

tingxuanp committed:
Fix shellscript test bug

tingxuanp committed:
Fix shellscript bug

tingxuanp committed:
Remove UI test folder

tingxuanp committed:
Remove redirection style automated testing

tingxuanp committed:
Update README.md
